plotly_resampler/aggregation/aggregation_interface.py

```diff
@@ -4,169 +4,260 @@
+            return y_agg, idxs
+
+        # An array filled with 1s and 2s, where 2 indicates a large gap mask
+        # (i.e., that index will be repeated twice)
+        repeats = np.ones(x_agg.shape, dtype="int") + gap_mask
+
+        # use the repeats to expand the idxs, and agg_y array
+        idx_exp_nan = np.repeat(idxs, repeats)
+        y_agg_exp_nan = np.repeat(y_agg, repeats)
+
+        # only float arrays can contain NaN values
+        if issubclass(y_agg_exp_nan.dtype.type, np.integer) or issubclass(
+            y_agg_exp_nan.dtype.type, np.bool_
+        ):
+            y_agg_exp_nan = y_agg_exp_nan.astype("float")
+
+        # Set the NaN values
+        # We add the gap index offset (via the np.arange) to the indices to account for
+        # the repeats (i.e., expanded y_agg array).
+        y_agg_exp_nan[np.where(gap_mask)[0] + np.arange(gap_mask.sum())] = None
 
-    def _supports_dtype(self, s: pd.Series):
+        return y_agg_exp_nan, idx_exp_nan
+
+    @staticmethod
+    def _supports_dtype(arr: np.ndarray, dtype_regex_list: Optional[List[str]] = None):
         # base case
-        if self.dtype_regex_list is None:
+        if dtype_regex_list is None:
             return
 
-        for dtype_regex_str in self.dtype_regex_list:
-            m = re.compile(dtype_regex_str).match(str(s.dtype))
+        for dtype_regex_str in dtype_regex_list:
+            m = re.compile(dtype_regex_str).match(str(arr.dtype))
             if m is not None:  # a match is found
                 return
         raise ValueError(
-            f"{s.dtype} doesn't match with any regex in {self.dtype_regex_list}"
+            f"{arr.dtype} doesn't match with any regex in {dtype_regex_list}"
         )
 
-    @staticmethod
-    def _calc_med_diff(s: pd.Series) -> Tuple[float, np.ndarray]:
-        # ----- divide and conquer heuristic to calculate the median diff ------
-        # remark: thanks to the prepend -> s_idx_diff.shape === len(s)
-        siv = s.index.values
-        s_idx_diff = np.diff(s.index.values, prepend=siv[0])
 
-        # To do so - use a quantile-based (median) approach where we reshape the data
-        # into `n_blocks` blocks and calculate the min
-        n_blcks = 128
-        if s.shape[0] > 5 * n_blcks:
-            blck_size = s_idx_diff.shape[0] // n_blcks
+class DataAggregator(AbstractAggregator, ABC):
+    """Implementation of the AbstractAggregator interface for data aggregation.
 
-            # convert the index series index diff into a reshaped view (i.e., sid_v)
-            sid_v: np.ndarray = s_idx_diff[: blck_size * n_blcks].reshape(n_blcks, -1)
+    A data aggregator is an aggregator that aggregates the data, and thus doesn't select
+    data points.
+    Concrete implementations of this class must implement the `_aggregate` method, and
+    have full responsibility on how they deal with other high-frequency properties, such
+    as `hovertext`, `marker_size`, 'marker_color`, etc ...
```

Comment:
Can we add two points to this documentation:
1) When to implement this class (and specifically if an end-user should)
and
2) What would an example implementation look like (briefly described or referenced)